src/PKSim.Presentation/Mappers/IndividualToIndividualBuildingBlockMapper.cs

```diff
@@ -0,0 +1,80 @@
+         ILazyLoadTask lazyLoadTask, ICalculationMethodToCategoryCalculationMethodDTOMapper calculationMethodDTOMapper) : base(objectBaseFactory, entityPathResolver, applicationConfiguration, lazyLoadTask)
+      {
+         _calculationMethodDTOMapper = calculationMethodDTOMapper;
+      }
+
+      protected override IReadOnlyList<IParameter> AllParametersFor(Individual sourcePKSimBuildingBlock)
+      {
+         return sourcePKSimBuildingBlock.GetAllChildren<IParameter>().Where(x => x.GroupName != CoreConstants.Groups.RELATIVE_EXPRESSION).ToList();
+      }
+
+      public override IndividualBuildingBlock MapFrom(Individual input)
+      {
+         var buildingBlock = base.MapFrom(input);
+         buildingBlock.Icon = input.Icon;
+
+         addOriginDataToBuildingBlock(buildingBlock, Assets.PKSimConstants.UI.DiseaseState, input.OriginData.DiseaseState?.DisplayName);
+         addOriginDataToBuildingBlock(buildingBlock, Assets.PKSimConstants.UI.Species, input.Species?.DisplayName);
+         addOriginDataToBuildingBlock(buildingBlock, Assets.PKSimConstants.UI.DiseaseState, input.OriginData.DiseaseState?.DisplayName);
+         addOriginDataToBuildingBlock(buildingBlock, Assets.PKSimConstants.UI.Population, input.OriginData.Population?.DisplayName);
+         addOriginDataToBuildingBlock(buildingBlock, Assets.PKSimConstants.UI.Gender, input.OriginData.Gender?.DisplayName);
+         addOriginDataToBuildingBlock(buildingBlock, Assets.PKSimConstants.UI.Age, input.OriginData.Age);
+         addOriginDataToBuildingBlock(buildingBlock, Assets.PKSimConstants.UI.GestationalAge, input.OriginData.GestationalAge);
+         addOriginDataToBuildingBlock(buildingBlock, Assets.PKSimConstants.UI.Height, input.OriginData.Height);
+         addOriginDataToBuildingBlock(buildingBlock, Assets.PKSimConstants.UI.BMI, input.OriginData.BMI);
+         addOriginDataToBuildingBlock(buildingBlock, Assets.PKSimConstants.UI.Weight, input.OriginData.Weight);
+         addOriginDataToBuildingBlock(buildingBlock, Assets.PKSimConstants.UI.Population, input.OriginData.Population?.DisplayName);
+
+         input.OriginData.AllCalculationMethods().MapAllUsing(_calculationMethodDTOMapper)
+            .Each(x => addOriginDataToBuildingBlock(buildingBlock, x.DisplayName, x.CategoryItem.DisplayName));
+
+         buildingBlock.OriginData.ValueOrigin = input.OriginData.ValueOrigin.Clone();
+
+         return buildingBlock;
+      }
+
+      private void addOriginDataToBuildingBlock(IndividualBuildingBlock buildingBlock, string key, OriginDataParameter parameter)
+      {
+         if (parameter == null)
+            return;
+
+         addOriginDataToBuildingBlock(buildingBlock, keyForOriginDataParameter(key, parameter), $"{parameter.Value} {parameter.Unit}");
+      }
+
+      private string keyForOriginDataParameter(string key, OriginDataParameter parameter)
+      {
+         return string.IsNullOrEmpty(parameter.Name) ? key : parameter.Name;
```

Comment:
I don't quite get this. How can you have  a parameter without a name? and if this is the case, why add the it at all ?

Reply:
I don't know how the OriginDataParameters are created for the individual, but many do not have the Name property set.

![image](https://user-images.githubusercontent.com/261477/210537884-82c3c942-c23c-4636-a1ee-a6438357feb3.png)